### ISSUE TITLE
Misleading error message for local function

### ISSUE THREAD

**Issue by controlflow** [dotnet/roslyn#18555]
**Version Used**: 
2.0

**Steps to Reproduce**:

Compile the following code:
```c#
public void M() {
   void Local()
}
```

**Expected Behavior**:

> error CS0XXX: local function 'Local()' must declare a body

**Actual Behavior**:

> error CS0501: 'Local()' must declare a body because it is not marked abstract, extern, or partial

Note that you neither can mark local function with `abstract`, `extern` or `partial` modifier

**Comment by khyperia** [comment]
Duplicate of #14900
